ortools/xpress/parse_header_xpress.py

```diff
@@ -36,23 +37,142 @@ def __init__(self):
+                                   "XPRS_COMPUTEMATX_IISMAXTIME", "XPRS_BARREFITER", "XPRS_COMPUTELOG",
+                                   "XPRS_SIFTPRESOLVEOPS", "XPRS_ESCAPENAMES", "XPRS_IOTIMEOUT", "XPRS_MAXSTALLTIME",
+                                   "XPRS_AUTOCUTTING", "XPRS_EXTRAELEMS", "XPRS_EXTRAPRESOLVE", "XPRS_EXTRASETELEMS",
+                                   "XPRS_LPOBJVAL", "XPRS_MIPOBJVAL", "XPRS_BESTBOUND", "XPRS_OBJRHS", "XPRS_OBJSENSE",
+                                   "XPRS_ROWS", "XPRS_SIMPLEXITER", "XPRS_LPSTATUS", "XPRS_MIPSTATUS", "XPRS_NODES",
+                                   "XPRS_COLS", "XPRS_LP_OPTIMAL", "XPRS_LP_INFEAS", "XPRS_LP_UNBOUNDED",
+                                   "XPRS_MIP_SOLUTION", "XPRS_MIP_INFEAS", "XPRS_MIP_OPTIMAL", "XPRS_MIP_UNBOUNDED",
+                                   "XPRS_OBJ_MINIMIZE", "XPRS_OBJ_MAXIMIZE", "XPRS_L1CACHE"}
+        self.__missing_required_defines = self.__required_defines
+        self.__required_functions = {"XPRScreateprob", "XPRSdestroyprob", "XPRSinit", "XPRSfree", "XPRSgetlicerrmsg",
+                                     "XPRSlicense", "XPRSgetbanner", "XPRSgetversion", "XPRSsetdefaultcontrol",
+                                     "XPRSsetintcontrol", "XPRSsetintcontrol64", "XPRSsetdblcontrol",
+                                     "XPRSsetstrcontrol", "XPRSgetintcontrol", "XPRSgetintcontrol64",
+                                     "XPRSgetdblcontrol", "XPRSgetstringcontrol", "XPRSgetintattrib",
+                                     "XPRSgetdblattrib", "XPRSloadlp", "XPRSloadlp64", "XPRSgetobj", "XPRSgetrhs",
+                                     "XPRSgetrhsrange", "XPRSgetlb", "XPRSgetub", "XPRSgetcoef", "XPRSaddrows",
+                                     "XPRSdelrows", "XPRSaddcols", "XPRSdelcols", "XPRSchgcoltype", "XPRSloadbasis",
+                                     "XPRSpostsolve", "XPRSchgobjsense", "XPRSgetlasterror", "XPRSgetbasis",
+                                     "XPRSwriteprob", "XPRSgetrowtype", "XPRSgetcoltype", "XPRSgetlpsol",
+                                     "XPRSgetmipsol", "XPRSchgbounds", "XPRSchgobj", "XPRSchgcoef", "XPRSchgmcoef",
+                                     "XPRSchgrhs", "XPRSchgrhsrange", "XPRSchgrowtype", "XPRSsetcbmessage", "XPRSminim",
+                                     "XPRSmaxim"}
+        self.__missing_required_functions = self.__required_functions
+
+    def should_define_be_imported(self, symbol):
+        return symbol in self.__required_defines
+
+    def should_fun_be_imported(self, name):
+        return name in self.__required_functions
 
     def write_define(self, symbol, value):
+        if not self.should_define_be_imported(symbol):
+            print('skipping ' + symbol)
+            return
+
         self.__header += f'#define {symbol} {value}\n'
+        self.__missing_required_defines.remove(symbol)
```

Comment:
1. Remove `return` statement, use `if..else`. 
2. Remove `not`, invert content of `if` and `else` blocks.